molecule/default/tests/test_cinder_lvm.py

```
@@ -1,47 +1,56 @@
-def test_cinder_lvs_volume_on_node(host):
-    # get list of volumes and associated hosts from utility container
-    cmd = ("{} cinder list --all-t --fields "
-           "os-vol-host-attr:host,status '".format(os_pre))
-    vol_table = host.run(cmd).stdout
-    vol_hosts = helpers.parse_table(vol_table)[1]
-    for vol, chost, status in vol_hosts:
-        print vol, chost, status
-        if status != 'available':
-            continue
+def test_cinder_lvs_volume_on_node(os_api_conn, host):
+    """Test to verify that a volume associated to a cinder host can be seen
+        on that cinder host as a logical volume. If there is any snapshot
+        created from the volume, then the snapshot is only seen on the cinder
+        host that has that volume.
+
+    Args:
+        os_api_conn (openstack.connection.Connection): An authorized API
+            connection to the 'default' cloud on the OpenStack infrastructure.
+        host (testinfra.host.Host): Testinfra host fixture.
+    """
+    # get VolumeDetail
+    vol_details = os_api_conn.block_storage.volumes()
+    # get SnapshotDetails
+    snapshot_details = os_api_conn.block_storage.snapshots()
+
+    for vol in vol_details:
+        chost = vol.to_dict()['host']
         chost = chost.split('@')[0].split('.')[0]
-        # VOLEXISTS test
-        cmd = "{} {} lvs | grep volume-{}".format(ssh_pre, chost, vol)
-        host.run_expect([0], cmd)
-        cmd = ("{} cinder snapshot-list"
-               " --all- --volume-id={} '".format(os_pre, vol))
-        snap_table = host.run(cmd).stdout
-        snaps = helpers.parse_table(snap_table)[1]
-        for snap in snaps:
-            # SNAPEXISTS test
-            snap_vol = snap[1]
-            cmd = ("{} {} lvs | "
-                   "grep _snapshot-{}".format(ssh_pre, chost, snap_vol))
+        vol_id = vol.to_dict()['id']
+        print("\nvolume ID in host {}: {}\n".format(chost, vol_id))
+
+        # Only run if current host is the chost
```

There is a class of bug that this design will mask. If the defined 'chost' is not actually in the 'cinder_hosts' group then it will be silently skipped.
This is an actual bug that has recently effected OSP: https://bugzilla.redhat.com/show_bug.cgi?id=1552214

This could be quickly checked by asserting that the chost is in the list of cinder_hosts:
```
assert chost in testinfra_hosts, 'The host associated with the cinder volume is not in the cinder_hosts group.'
```